pyccel/ast/numpyext.py

```diff
@@ -537,26 +537,40 @@ class NumpyLinspace(NumpyNewArray):
     Represents numpy.linspace.
 
     """
-    __slots__ = ('_index','_start','_stop','_size','_shape', '_rank')
-    _dtype     = NativeReal()
-    _precision = default_precision['real']
+    __slots__ = ('_dtype','_precision','_index','_start','_stop','_num','_shape', '_rank')
     _order     = 'F'
 
-    def __init__(self, start, stop, size):
-
+    def __init__(self, start, stop, num=None):
```

Comment:
Since the goal of this PR is to support `Numpy.linspace`, I think that we should support the same arguments as described in Numpy's documentation (https://numpy.org/doc/stable/reference/generated/numpy.linspace.html). The missing arguments are:

- `endpoint : bool, optional`
- `retstep  : bool, optional`
- `dtype : dtype, optional`
- `axis : int, optional`

It is going to be tricky to add `retstep`, as it changes the number of outputs of this function, so I suggest to create a separate issue about that. But the other three arguments (`endpoint`, `dtype`, `axis`) should definitely be added to this PR.